metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -265,10 +304,71 @@ def _extract_record(self, topic: str) -> Iterable[MetadataWorkUnit]:
                 self.report.report_workunit(wu)
                 yield wu
 
+    def build_custom_properties(self, topic_detail, extra_topic_config):
+
```

Comment:
is topic_detail ever going to be None? Do we need to handle this case?

Reply:
topic_detail is supposed to be instance of TopicMetadata. However I have added handling None here to be safe.